devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -102,16 +143,21 @@ private handleReportAttributeMessage(String description) {
  * @param volts Battery voltage in .1V increments
  */
 private handleBatteryEvent(volts) {
-	def descriptionText
+    def descriptionText
     if (volts == 0 || volts == 255) {
         log.debug "Ignoring invalid value for voltage (${volts/10}V)"
     }
     else {
-        def batteryMap = [28:100, 27:100, 26:100, 25:90, 24:90, 23:70,
-                          22:70, 21:50, 20:50, 19:30, 18:30, 17:15, 16:1, 15:0]
+        def batteryMap = getBatteryMap()
         def minVolts = 15
-        def maxVolts = 28
-
+        def maxVolts
+        
+        if (isVision()) {
+            maxVolts = 29
+        } else {
+            maxVolts = 28
+        }
+               
         if (volts < minVolts)
```

Comment:
Please, could You add parentheses to this block ?

Reply:
OK! I will add parentheses to this block.